plugins/stack-overflow-backend/src/search/StackOverflowQuestionsCollatorFactory.ts

```diff
@@ -93,12 +100,16 @@ export class StackOverflowQuestionsCollatorFactory
         `No stackoverflow.baseUrl configured in your app-config.yaml`,
       );
     }
+
     const params = qs.stringify(this.requestParams, {
       arrayFormat: 'comma',
       addQueryPrefix: true,
     });
 
-    const res = await fetch(`${this.baseUrl}/questions${params}`);
+    const apiKey = this.config.getOptionalString('stackoverflow.apiKey');
```

Comment:
This config can be updated too:
https://github.com/backstage/backstage/blob/713be06b2bd26a2460732ca2e7d523aab0aa9821/plugins/stack-overflow-backend/config.d.ts